File: java/parser/src/main/java/gov/lbl/parser/parser/Der_class_specifierVisitor.java
package gov.lbl.parser.parser;

import java.util.List;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Comment;
import gov.lbl.parser.domain.Der_class_specifier;
import gov.lbl.parser.domain.Der_class_specifier_value;
import gov.lbl.parser.domain.Name;

import static java.util.stream.Collectors.toList;

public class Der_class_specifierVisitor extends modelicaBaseVisitor<Der_class_specifier> {
    @Override
    public Der_class_specifier visitDer_class_specifier(modelicaParser.Der_class_specifierContext ctx) {
        List<String> idents = ctx.IDENT()
            .stream()
            .map(IDENT -> IDENT.getText())
            .collect(toList());
        String identifier = idents.get(0);

        List<String> identifiers = idents.subList(1, idents.size());
        NameVisitor nameVisitor = new NameVisitor();
        Name type_specifier = ctx.name().accept(nameVisitor);
        CommentVisitor commentVisitor = new CommentVisitor();
        Comment comment = ctx.comment().accept(commentVisitor);

        Der_class_specifier_value der_class_specifier_value = new Der_class_specifier_value(type_specifier, identifiers, comment);
        return new Der_class_specifier(identifier, der_class_specifier_value);
    }    
}

File: jsParser/parser/Der_class_specifierVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Der_class_specifier = require('../domain/Der_class_specifier')
const Der_class_specifier_value = require('../domain/Der_class_specifier_value')

const NameVisitor = require('./NameVisitor')
// const CommentVisitor = require('./CommentVisitor');

class Der_class_specifierVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitDer_class_specifier (ctx) {
    let identifier = ''
    var der_class_specifier_value = null
    if (ctx.IDENT()) {
      identifier = ctx.IDENT(0).getText()
    }

    let type_specifier = null
    let identifiers = []
    let comment = null

    if (ctx.name()) {
      const nameVisitor = new NameVisitor.NameVisitor()
      type_specifier = nameVisitor.visitName(ctx.name())
    }
    if (ctx.IDENT().length > 1) {
      ctx.IDENT().forEach(ident => {
        identifiers.push(ident.getText())
      })
      identifiers = identifiers.slice(1)
    }
    if (ctx.comment()) {
      const commentVisitor = new CommentVisitor.CommentVisitor()
      comment = commentVisitor.visitComment(ctx.comment())
    }
    var der_class_specifier_value = new Der_class_specifier_value.Der_class_specifier_value(type_specifier, identifiers, comment)

    return new Der_class_specifier.Der_class_specifier(identifier, der_class_specifier_value)
  }
};

Der_class_specifierVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitDer_class_specifier = this.visitDer_class_specifier
exports.Der_class_specifierVisitor = Der_class_specifierVisitor
